File: backend/src/main/java/com/backend/ord/domain/dto/LanguageProficiencyDTO.java
package com.backend.ord.domain.dto;

import com.backend.ord.domain.dto.abstracts.DTOBase;
import com.backend.ord.enums.Language.LanguageName;
import com.backend.ord.enums.Language.LanguageProficiencyLevel;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class LanguageProficiencyDTO extends DTOBase {
    private LanguageName language;
    private LanguageProficiencyLevel proficiency;

    private UserDTO user;
}

File: backend/src/main/kotlin/com/backend/ord/domain/persistence/dto/LanguageProficiencyDTO.kt
package com.backend.ord.domain.persistence.dto

import com.backend.ord.core.user.model.UserDTO
import com.backend.ord.enums.persistence.language.LanguageName
import com.backend.ord.enums.persistence.language.LanguageProficiencyLevel
import java.time.Instant
import java.util.*

data class LanguageProficiencyDTO(
    val id: UUID = UUID.randomUUID(),

    var language: LanguageName,
    var proficiency: LanguageProficiencyLevel,
    var generativeContentLanguage: LanguageName,

    var user: UserDTO,

    val createdAt: Instant = Instant.now(),
    var updatedAt: Instant = Instant.now(),
)
